packages/bsky/src/notifications.ts

```diff
@@ -0,0 +1,239 @@
+import axios from 'axios'
+import Database from './db'
+import { Notification } from './db/tables/notification'
+import { AtUri } from '@atproto/api'
+import { Insertable } from 'kysely'
+import logger from './indexer/logger'
+
+type Platform = 'ios' | 'android' | 'web'
+type PushNotification = {
+  tokens: string[]
+  platform: 1 | 2 // 1 = ios, 2 = android
+  title: string
+  message: string
+  topic: string
+  data?: {
+    [key: string]: string
+  }
+}
+type InsertableNotif = Insertable<Notification>
+
+export class NotificationServer {
+  constructor(public db: Database, public pushEndpoint?: string) {}
+
+  async getUserTokens(did: string) {
+    const userTokens = await this.db.db
+      .selectFrom('notification_push_token')
+      .where('did', '=', did)
+      .selectAll()
+      .execute()
+
+    return userTokens
+  }
+
+  async prepareNotifsToSend(notifications: InsertableNotif[]) {
+    const notifsToSend: PushNotification[] = []
+
+    for (const notif of notifications) {
+      const { did: userDid } = notif
+      const userTokens = await this.getUserTokens(userDid)
```

Comment:
would be cool to get notifs for all relevant users in a single request instead of multiple requests

`.where('did', 'in', dids)`

Reply:
I assume by this you mean changing the `getUserTokens` function to have `.where('did', 'in', dids)` so we can grab all the dids we need in one request instead of many?